prow/plugins/config.go

```diff
@@ -730,39 +750,57 @@ func (r RequireMatchingLabel) Describe() string {
 	return str.String()
 }
 
-// ApproveFor finds the Approve for a repo, if one exists.
-// Approval configuration can be listed for a repository
-// or an organization.
-func (c *Configuration) ApproveFor(org, repo string) *Approve {
-	fullName := fmt.Sprintf("%s/%s", org, repo)
-
-	a := func() *Approve {
-		// First search for repo config
-		for _, approve := range c.Approve {
-			if !sets.NewString(approve.Repos...).Has(fullName) {
-				continue
+func oldToNewApprove(old DeprecatedApprove) *Approve {
+	a := Approve{
+		IgnoreReviewState:   old.IgnoreReviewState,
+		IssueRequired:       old.IssueRequired,
+		LgtmActsAsApprove:   old.LgtmActsAsApprove,
+		RequireSelfApproval: old.RequireSelfApproval,
+		CommandHelpLink:     old.CommandHelpLink,
+		PrProcessLink:       old.PrProcessLink,
+	}
+	return &a
+}
+
+func oldToNewApproveConfig(old []DeprecatedApprove) ApproveConfigTree {
```

Comment:
I just don't like this so much... and I should probably try to move it to the generic class by passing `oldToNewApprove` as an argument.
What do you think?